File: CS/Q132330/WindowsApplication1/Form1.cs
// Developer Express Code Central Example:
// How to implement custom Recurrence form with hourly recurrence
// 
// This sample illustrates how to create custom Recurrence form, which contains
// Hourly recurrence pattern
// 
// You can find sample updates and versions for different programming languages here:
// http://www.devexpress.com/example=E558

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraScheduler.UI;
using DevExpress.XtraScheduler;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void schedulerControl1_EditAppointmentFormShowing(object sender, AppointmentFormEventArgs e)
        {
            MyAppointmentForm frm = new MyAppointmentForm(sender as SchedulerControl, e.Appointment, e.OpenRecurrenceForm);
            e.DialogResult = frm.ShowDialog();
            e.Handled = true;
        }
    }
    public class MyAppointmentForm : AppointmentForm
    {
        public MyAppointmentForm(SchedulerControl control, Appointment apt)
            : base(control, apt)
        {
        }
        public MyAppointmentForm(SchedulerControl control, Appointment apt,bool openRecurrenceForm)
            : base(control, apt, openRecurrenceForm)
        {
        }
        protected override Form CreateAppointmentRecurrenceForm(Appointment patternCopy, FirstDayOfWeek firstDayOfWeek)
        {
            XtraForm1 frm = new XtraForm1(patternCopy, firstDayOfWeek, this.Controller);
            frm.LookAndFeel.ParentLookAndFeel = this.LookAndFeel;
            frm.ShowExceptionsRemoveMsgBox = this.Controller.AreExceptionsPresent();
            return frm;
        }
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraScheduler.UI
Imports DevExpress.XtraScheduler

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub schedulerControl1_EditAppointmentFormShowing(ByVal sender As Object, ByVal e As AppointmentFormEventArgs) Handles schedulerControl1.EditAppointmentFormShowing
			Dim frm As New MyAppointmentForm(TryCast(sender, SchedulerControl), e.Appointment, e.OpenRecurrenceForm)
			e.DialogResult = frm.ShowDialog()
			e.Handled = True
		End Sub
	End Class
	Public Class MyAppointmentForm
		Inherits AppointmentForm
		Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment)
			MyBase.New(control, apt)
		End Sub
		Public Sub New(ByVal control As SchedulerControl, ByVal apt As Appointment, ByVal openRecurrenceForm As Boolean)
			MyBase.New(control, apt, openRecurrenceForm)
		End Sub
        Protected Overrides Function CreateAppointmentRecurrenceForm(ByVal patternCopy As Appointment, ByVal firstDayOfWeek As DevExpress.XtraScheduler.FirstDayOfWeek) As Form
            Dim frm As New XtraForm1(patternCopy, firstDayOfWeek, Me.Controller)
            frm.LookAndFeel.ParentLookAndFeel = Me.LookAndFeel
            frm.ShowExceptionsRemoveMsgBox = Me.Controller.AreExceptionsPresent()
            Return frm
        End Function
	End Class
End Namespace